src/components/productDetails/list.jsx

```diff
@@ -78,6 +78,20 @@ const List = ({
     }
   }
 
+  function checkForMissingShownImage(productsImages, shownIndices, idx) {
+    // checks to see if results have come in yet and whether or not there are actual results to work with
+    if (filterForShownItems(productsImages, shownIndices)[idx] !== undefined) {
+      if (
+        filterForShownItems(productsImages, shownIndices)[idx].results.length >
+        0
```

Comment:
am i understanding correctly that the undefined will check if there is actual data, and the results.length with check if there are are are results in that existing data?

Reply:
yes that is 100% correct.